### PR TITLE
Set Valid JSON to a Consul Key

### LINKED ISSUE #12: Have a set command which writes values as JSON to Consul.

### PR DESCRIPTION
**Pull request by abhiyerra**
Closes #12

### CODE REVIEW

File: set.go [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+	if len(args) < 2 {
+		fmt.Println("jsonconsul set full/key/path \"true\"")
+		os.Exit(-1)
+	}
+
+	js.Key = args[0]
+	js.value = args[1]
+}
+
+func (js *JsonSet) lintedJson() []byte {
```
Comment (twoism) [inline set.go]: We should probably ensure the value is either a bool or float at this point as well.

**Comment by twoism** [comment]
Just one nit from me.

File: set.go [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+		if js.ExpectedType != Array {
+			return fmt.Errorf("Invalid type. Value is an array. Expected %s", js.expectedType())
+		}
+	case map[string]interface{}:
+		if js.ExpectedType != Object {
+			return fmt.Errorf("Invalid type. Value is an object. Expected %s", js.expectedType())
+		}
+	}
+
+	return nil
```
Comment (twoism) [inline set.go]: You dont need to return here.

Comment (parkr) [inline set.go]: You always have to return _something_, right? The Go compiler has yelled at me before if I don't include this line. Maybe it was in a different scenario?

Comment (abhiyerra) [inline set.go]: No you always need to return. Go compiler complains if there is no return.

File: set.go [hunk 3]
```diff
@@ -0,0 +1,147 @@
...
+}
+
+func (js *JsonSet) ParseFlags(args []string) {
+	flags := flag.NewFlagSet(Name, flag.ContinueOnError)
+
+	expectedType := flags.String("expected-type", "undefined", "What is the expected type for the value? bool, int, float, string, object, array")
```
Comment (twoism) [inline set.go]: Do we want to allow "undefined" as a type?

File: set.go (hunk 2 above)
Comment (twoism) [inline set.go]: ah, i see, you're right. i was hoping we could avoid returning nil but it seems you have to. nevermind.

File: set.go (hunk 3 above)
Comment (abhiyerra) [inline set.go]: Want to keep expected-type optional. Might have use cases outside of feature flags where that is valuable.

Comment (twoism) [inline set.go]: OK, makes sense.

**Comment by twoism** [comment]
shipit!

**Pull request merged**